src/z3c/relationfield/relation.py

```diff
@@ -73,10 +79,18 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
-    def __cmp__(self, other):
-        if other is None:
-            return cmp(self._sort_key(), None)
-        return cmp(self._sort_key(), other._sort_key())
+    def __lt__(self, other):
+        ''' Relations are sorted by default on a combination of the relation name,
+        the path of the object the relation is one and the path of the object
+        the relation is pointing to.
+        '''
+        if (self.from_attribute or '') < (other.from_attribute or ''):
+            return True
+        if (self.from_path or '') < (other.from_path or ''):
+            return True
+        if (self.to_path or '') < (other.to_path or ''):
+            return True
+        return False
 
     def _sort_key(self):
         return (self.from_attribute, self.from_path, self.to_path)
```

Comment:
If from_attribute/from_path/to_path can be None, won't this cause TypeErrors on Python 3?

Reply:
That's why I am adding `or ''`, to always compare str

Reply:
But not in `_sort_key()`.

If `_sort_key()` is no longer used, it should be removed.  If it is still used, it should handle sorting without TypeError when some of the attributes are None.